### ISSUE TITLE
Images not found if Flexible field is in a Repeater field

### ISSUE THREAD

**Issue by tmconnect** [BeAPI/bea-beautiful-flexible#7]
First of all: Great work!

But the images for the popup are not found if the Flexible field is in a Repeater field.

I found a solution for an existing scenario on my site, but I don't know if this is the correct way for all. I changed the `foreach ( $fields as $field )` loop in the `retrieve_flexible_keys` function to this:
````
foreach ( $fields as $field ) {
	if ( 'repeater' === $field['type'] ) {
		foreach ($field['sub_fields'] as $repeaterfield) {
			if ( 'flexible_content' === $repeaterfield['type'] ) {
				foreach ( $repeaterfield['layouts'] as $layout_field ) {
					$keys[ $layout_field['key'] ] = $layout_field['name'];
				}							
			}
		}
	} else if ( 'flexible_content' === $field['type'] ) {
		// Flexible is recursive structure with sub_fields into layouts
		foreach ( $field['layouts'] as $layout_field ) {
			if ( $keys [ $layout_field ] ) {
				continue;
			}
			$keys[ $layout_field['key'] ] = $layout_field['name'];
		}
	}
}
````
Hope, this helps for further development.

Thomas

**Comment by MaximeCulea** [comment]
Great ! I'll have a look closely soon then.

**Comment by MaximeCulea** [comment]
Hi @tmconnect, 
Sorry for the delay, I had some hollidays 🤗 

Could you please check that this [version](https://github.com/BeAPI/bea-beautiful-flexible/archive/issue-7.zip) include your changes and works well in your case ?

You will be kind, thank you 😄

**Comment by tmconnect** [comment]
Thanks for the new version, but I got an error - even if the flexible content field is not in repeater:
Illegal offset type in .../wp-content/plugins/bea-beautiful-flexible-issue-7/classes/main.php on line 85

After delving a bit deeper into your code, I found a much simpler solution for the implementation:

Because each popup item already has a data-layout attribute, which tells the name of the layout element. You don't have to get the names of all flexible layouts in an extra function.

The JS script only has to get this data-layout attribute and has to modify the acf-popup list. In addition I wrote some data attributes to the acf-popup <u> element for the title and subtitle for the popup. So the strings can be translated.

You can download my solution from my dropbox: [Link](https://www.dropbox.com/s/uzfnhgkjsntmmkh/bea-beautiful-flexible-2beta.zip?dl=0)

Let me know if this works for you.

Thomas

**Comment by MaximeCulea** [comment]
Hi @tmconnect,
It seems that this error was already present and now fixed with #10.
Except this error, everything was fine ? 
Please let me now in order to merge and release. 
Thx

**Comment by herewithme** [comment]
This plugin is deprecated.
I recommend you to use : https://wordpress.org/plugins/acf-extended/
